.github/workflows/pr-linting.yml

```diff
@@ -23,3 +23,5 @@ jobs:
         run: npm run lerna:bootstrap
       - name: Run linting of subpackages
         run: npm run lerna:lint -- --no-bail --concurrency 1 --stream --no-prefix
+      - name: Run linting of C++ code
+        run: for i in `find src -name "*.?pp" | grep -v catch`; do clang-format --dry-run --Werror $i; done
```

Comment:
I was wondering if we wanted to use the clang-format NPM package and add a script to the package running this command?

Also, isn't there a more compact way of running this on a directory of files?

Reply:
I guess using xargs will eliminate the for loop https://stackoverflow.com/a/36046965
Guess that's a matter of taste 🙂

Reply:
Ah yes I forgot about the clang-format package – I'll take a look and try that. I have no strong feelings about how we run it, I tend to use for loops because I always forget the xargs syntax haha, but it is maybe a bit neater so I'll change

Reply:
Probably a matter of taste ... I tend to forget the for-loop bash syntax 🙈

Reply:
As usual with shell, both are quite awkward syntaxes 😁

Reply:
Update to use the npm package which I agree is cleaner